src/main/scala/nl.knaw.dans.dccd.convert/DccdConvertExportApp.scala

```diff
@@ -15,10 +15,216 @@
+    new File(directoryName)
+      .listFiles
+      .filter(_.isDirectory)
+      .map(_.getName)
+  }
+
+  def getListOfFiles(dir: String): List[File] = {
+    val d = new File(dir)
+    if (d.exists && d.isDirectory) {
+      d.listFiles.filter(_.isFile).toList
+    }
+    else {
+      List[File]()
+    }
+  }
+
+  def getPath(dir: String): Path = {
+    Paths.get(configuration.properties.getString(dir))
+  }
+
+  val csvFormat: CSVFormat = CSVFormat.RFC4180.withHeader("DATASET", "DC_TITLE", "DEPOSITOR_ID", "DCX_CREATOR_ORGANIZATION", "DDM_CREATED", "DCT_RIGHTSHOLDER", "DC_DESCRIPTION", "DC_SUBJECT", "DCT_TEMPORAL", "DCX_SPATIAL_X", "DCX_SPATIAL_Y", "DC_IDENTIFIER", "DCX_RELATION_QUALIFIER", "DCX_RELATION_TITLE", "DCX_RELATION_LINK", "DCX_RELATION_QUALIFIER(2)", "DCX_RELATION_TITLE(2)", "DC_TYPE", "DC_LANGUAGE", "DDM_ACCESSRIGHTS", "DDM_AVAILABLE", "DDM_AUDIENCE").withDelimiter(',')
+  val out: Appendable = new StringBuffer()
+  val printer: CSVPrinter = csvFormat.print(out)
+
+  val csvPrinterToFile = new CSVPrinter(new FileWriter("./data/dccdConvert" + ".csv"), csvFormat.withDelimiter(','))
+
+  private def parse(s: String) = fromSource(fromString(s), preserveWS = true).element(TopScope)
+
+  private def parseNoWS(s: String) = fromSource(fromString(s), preserveWS = false).element(TopScope)
+
+  var dirr: String = _
+  var dirr2: String = _
+  var dirPath: Path = _
+  var dirList: List[String] = _
+  var fileList: List[File] = _
+  var file2: File = _
+
+  var metadata: Elem = _
+  var parsedMetadata: NodeSeq = _
+  var user: Elem = _
+  var parsedUser: NodeSeq = _
+
+  var dataset: String = _
+  var dcTitle: String = _
+  var depositorId: String = _
+  var dcxCreatorOrganization: String = _
+
+  val Format = new SimpleDateFormat("yyyy-MM-dd")
+
+  var date: String = _
+  var parsedDate: AnyRef = _
+  var ddmCreated: String = _
+
+  var dctRightsHolder: String = _
+  var dcDescription: String = _
+  var dcSubject: String = _
+  var dcElementType: String = _
+  var dcObjectType: String = _
+  var dcTypeMetadata: String = _
+  var dcCategory: String = _
+  var dcTaxon: String = _
+
+  var range: String = _
+
+  var dctTemporal: String = _
+
+  var dcxSpatialX: String = _
+  var dcxSpatialY: String = _
+
+  var dcIdentifierSid: String = _
+  var dcIdentifierIdentifier: String = _
+
+  var dcxRelationQualifier: String = _
+  var dcxRelationTitle: String = _
+  var dcxRelationLink: String = _
+
+  var dcxRelationQualifier_2: String = _
+  var dcxRelationTitle_2: String = _
+
+  var dcType: String = _
+
+  var language: String = _
+  var isoDefault: Locale = _
+  var languageISO3: String = _
+
+  var ddmAccessRights: String = _
+  var now: Date = _
+  var ddmAvailable: String = _
+  var ddmAudience: String = _
+
+
+  def createCsvReportFromXmlFiles(dir: String): String = {
+    dirPath = getPath(dir)
+    dirList = getListOfSubDirectories(getPath(dir).toString).toList
+    dirList.foreach { i =>
```

Comment:
This is your loop for 'each' project. Refactor it and put the 'single' project work in a separate function. Then the outermost function looping over all projects becomes more readable.